sources/packages/backend/apps/queue-consumers/src/services/application/application.service.ts

```diff
@@ -153,4 +153,67 @@ export class ApplicationService {
         .getMany()
     );
   }
+
+  /**
+   * Retrieves applications eligible for a specific notification (likely related to disability status and PD/PPD status).
+   * This method applies several criteria to filter eligible applications:
+   * - Study end date is at least 8 weeks in the future
+   * - Student's disability status is not 'PD' or 'PPD'
+   * - The assessment's workflow data indicates a positive PD/PPD status
+   * - The application is not archived
+   * - There's a pending disbursement schedule
+   * - No notification of this type (messageId 30) has been sent for this assessment
+   * @returns An array of eligible applications with relevant details for notification.
+   */
+  async getEligibleApplicationsForNotification(): Promise<Application[]> {
+    const eightWeeksFromNow = new Date();
+    eightWeeksFromNow.setDate(eightWeeksFromNow.getDate() + 56); // 8 weeks * 7 days
+
+    return this.applicationRepo
+      .createQueryBuilder("application")
+      .select([
+        "application.applicationNumber",
+        "users.firstName",
+        "users.lastName",
+        "users.email",
+        "users.id",
+        "student_assessments.id",
+        "students.disabilityStatus",
+      ])
+      .innerJoin("application.currentAssessment", "student_assessments")
+      .innerJoin("application.student", "students")
+      .innerJoin("students.user", "users")
+      .innerJoin("student_assessments.offering", "epo")
+      .innerJoin("student_assessments.disbursementSchedules", "ds")
+      .where("epo.studyEndDate >= :eightWeeksFromNow", { eightWeeksFromNow })
+      .andWhere("students.disabilityStatus NOT IN ('PD', 'PPD')")
+      .andWhere(
+        "json_extract_path_text(student_assessments.workflow_data::json, 'calculatedData', 'pdppdStatus') = 'true'",
+      )
+      .andWhere("application.isArchived = false")
+      .andWhere("ds.disbursementScheduleStatus = 'Pending'")
+      .andWhere((qb) => {
+        const subQuery = qb
+          .subQuery()
+          .select("1")
+          .from("notifications", "n")
+          .where("n.notification_message_id = 30")
+          .andWhere(
+            "json_extract_path_text(n.metadata::json, 'assessmentId') = CAST(student_assessments.id AS TEXT)",
+          )
+          .getQuery();
+        return "NOT EXISTS (" + subQuery + ")";
+      })
+      .andWhere((qb) => {
+        const subQuery = qb
+          .subQuery()
+          .select("1")
+          .from("disbursement_schedules", "ds2")
+          .where("ds2.studentAssessment = student_assessments.id")
+          .andWhere("ds2.disbursementDate < ds.disbursementDate")
+          .getQuery();
+        return "NOT EXISTS (" + subQuery + ")";
+      })
```

Comment:
Is this `NOT EXISTS` a solution to eliminate the possible duplication caused by the inner join with the schedules?
https://github.com/bcgov/SIMS/pull/3759/files#r1786933162

If yes, the inner join can be removed and the pending schedules can be checked with an "exists". Does it make sense?

Reply:
Please check the [shared query](https://github.com/bcgov/SIMS/pull/3759/files#r1788340560) with the implemented example.